azurelinuxagent/ga/exthandlers.py

```diff
@@ -76,6 +76,11 @@
 
 NUMBER_OF_DOWNLOAD_RETRIES = 5
 
+DISABLE_FAILED = "DISABLE_FAILED"
+UNINSTALL_FAILED = "UNINSTALL_FAILED"
+EXTENSION_PATH = "EXTENSION_PATH"
+EXTENSION_VERSION = "EXTENSION_VERSION"
```

Comment:
There is already a VERSION env variable that gets set in ExtHandlerInstance.update

Reply:
Yes but that's only there for when the update command is called. This would be available for every command. I didnt wanna remove VERSION variable because customers might have some dependencies on it

Reply:
yes, we should not remove it. It is odd. though, that VERSION and EXTENSION_VERSION are handled differently in the code (they have the same value and semantics, right?

Reply:
So version is passed as a parameter to the update() command IF we're downgrading the extension.
`old_ext_handler_i.update(version=ext_handler_i.ext_handler.properties.version)`

 Otherwise it uses the same semantics - 
```python
    def update(self, version=None):
        if version is None:
            version = self.ext_handler.properties.version
```